### ISSUE TITLE
cd script fixes

### ISSUE THREAD

**Issue by edublancas** [ploomber/dummy#36]
we were able to do an end-to-end test, but there are a few things that are still broken/missing:

## lost git tag

we are creating the tag in a temporary branch  (the one with the output of `pkgmt version`); this branch is used to open a PR. However, when merged, the tag isn't copied over to the main branch ([see this](https://stackoverflow.com/q/12278660/709975)) and it stays in the temporary branch. So while the [tag exists](https://github.com/ploomber/dummy/commit/d9aaaadd81f56efc819a1eb39c4c14e7b02c7ecf), it wont be part of the main branch (that's why we see the "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository." message)

to fix it:

- modify pkgmt so it doesn't create a tag when running `pkgmt version`
- a [github action](https://itnext.io/creating-a-github-action-to-tag-commits-2722f1560dec) should look at the main branch and add the tag to the correct commit 

## grabbing the proper release commit

once we merge a feature PR, the automatically created PR contains two commits: one with the version to be released and another one with the bumped-up development version. The problem is that the reviewer might accidentally squash the commits, causing us to lose the release commit. Furthermore, when pushing multiple commits at once, github actions are only triggered for the latest commit. hence; we are unable to trigger anything for the commit with the stable version.

to fix it: we can trigger a github action when approving the feature PR that will add a new commit to it with the stable version. Hence, merging the feature PR will already contain the stable version.

Then a second action can look at commits in main (possibly look at the commit message), trigger the release, and open a new PR with the new development version.

**Comment by edublancas** [comment]
I made some progress in addressing some of the blockers we encountered. I modified pkgmt and created a [workflow](https://github.com/ploomber/dummy/blob/main/.github/workflows/pr-comment.yml) that adds a new commit to a PR when you comment a keyword - the problem is that this will *trigger a github actions job on every PR comment*, which is pretty wasteful.  but I'm unsure if there are alternatives

so the new workflow is as follows:

1. contributor submits a PR
2. reviewer approves the PR, and comments with a special keyword
3. bot adds a new commit that updates changelog and version from a development to a stable one (it can determine if it's a minor or major upgrade by looking at the changelog entries)
4. reviewer merges the PR

A second github action looks for commits in the main branch that met specific criteria* and:

1. create a tag and push it to the repository (can we do it? Wouldn't this fail because of branch protections?)
2. uploads to PyPI
3. creates a second PR with to upgrade to a new development version (we cannot commit directly due to branch protections)

*unsure about these criteria (maybe the commit message + checking who approved the PR merge?). There are some security aspects here since this trigger will publish to PyPI.
